### PR TITLE
fix potential crash in rgw_kafka

### PR DESCRIPTION
**Pull request by yuvalif**
issues were detected by pvs-studio static analyzer

Signed-off-by: Yuval Lifshitz <ylifshit@redhat.com>

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
jenkins test make check

File: src/rgw/rgw_kafka.cc [hunk 1]
```diff
@@ -107,7 +107,7 @@ struct connection_t {
   // ctor for setting immutable values
   connection_t(CephContext* _cct, const std::string& _broker, bool _use_ssl, bool _verify_ssl, 
           const boost::optional<const std::string&>& _ca_location,
-          const std::string& _user, const std::string& _password) :
+          const std::string& _user, const std::string& _password) : status(STATUS_OK),
```
Comment (cbodley) [inline src/rgw/rgw_kafka.cc:110]: siunce you're already using member initializers for other stuff:
```diff
   uint64_t delivery_tag = 1;
-  int status;
+  int status = STATUS_OK;
   mutable std::atomic<int> ref_count = 0;
```

**Review by cbodley — APPROVED** [review]

File: src/rgw/rgw_kafka.cc (hunk 1 above)
Comment (yuvalif) [reply src/rgw/rgw_kafka.cc]: fixed

**Comment by yuvalif** [comment]
teuthology [results](http://pulpito.front.sepia.ceph.com/yuvalif-2020-10-08_04:57:15-rgw-wip-yuval-static-analysis-distro-basic-smithi/) has 9 failures:
- `s3tests_boto3.functional.test_s3.test_bucket_listv2_delimiter_basic`: (was a test issue)
- 2 failures in: `test_get_object_with_x_object_manifest`: (happened also on master)
- failure with:
```
Health check failed: Reduced data availability: 1 pg peering (PG_AVAILABILITY)"
```
- barbican "create secret" issue: (happened also on master)
```
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/contextutil.py", line 31, in nested
    vars.append(enter())
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-yuval-lua-scripting/qa/tasks/barbican.py", line 360, in create_secrets
    raise Exception("Cannot create secret")
Exception: Cannot create secret
```
- 2 multisite tests failed on `test_datalog_autotrim`
- 2 multisite tests failed on `test_datalog_autotrim` as well as other tests:
  - http://qa-proxy.ceph.com/teuthology/yuvalif-2020-10-08_04:57:15-rgw-wip-yuval-static-analysis-distro-basic-smithi/5505756/teuthology.log
  - http://qa-proxy.ceph.com/teuthology/yuvalif-2020-10-08_04:57:15-rgw-wip-yuval-static-analysis-distro-basic-smithi/5505726/teuthology.log

**Comment by yuvalif** [comment]
recent teuthology results see here: https://github.com/ceph/ceph/pull/37534#issuecomment-706658582

**Pull request merged**